src/geometric_algebra/nursery/chisolm.lean

```diff
@@ -69,6 +69,28 @@ local infix `*₊`:75 := sym_prod
 
 local infix `*₊ᵥ`:75 := sym_prod_vec
 
+
+def is_orthogonal (a : G₁) (b : G₁) : Prop := sym_prod_vec a b = 0
+
+theorem zero_is_orthogonal (a : G₁) : is_orthogonal 0 a := begin
+  unfold is_orthogonal,
+  unfold sym_prod_vec,
+  unfold prod_vec,
+  simp
+end
+
+class inductive blade (b : G): nat → Type u
+| scalar :
+  -- equals a scalar
+  (∃ (k: G₀),
+   b = fₛ k)
+  → blade 0
+| graded {n : ℕ} :
+  -- or a product of orthogonal vectors
+  (∃ (v : {l : vector G₁ (n + 1) // l.val.pairwise (λ a b, is_orthogonal a b ∧ a ≠ b)}),
+   b = list.foldl (λ (b : G) (a : G₁), fᵥ a * b) (fᵥ v.val.head) v.val.tail.val)
```

Comment:
We could do two things:

1. use https://github.com/leanprover-community/mathlib/blob/master/src/algebra/big_operators.lean instead of the foldl to make it look like math formula more
2. have a lemma to do the unfold/simp stuff, so the actual proof can focus on the real goal

Reply:
See https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/iterating.20over.20subscripts.20.26.20generalizing.20products/near/201907701 for an example of 1

Reply:
Another approach would be to up-convert all the vectors to `G` and use `list.prod`. I suspect an explicit foldl might be easier to recourse over, but have little intuition there.

Reply:
What I had in mind was to first prove GA is a vector space, and reuse the theorems there to prove the stuff about basis, generated, span etc. I still trying to figure out how to construct stuff based on theorems.

As for graded algebra, previously I found [commutative differential graded algebras](https://gist.github.com/kbuzzard/f5ee35457c5d257ceec58c66d7da0c38) to be a good reference implementation. It seems that I don't really need the bases to define the grade operations.